src/utils/hooks/use-generate-pathname.js

```diff
@@ -15,59 +34,83 @@ const getCurrentActivePipeline = () => {
 export const useGeneratePathname = () => {
   const history = useHistory();
 
+  /**
+   * Updates the URL with search parameters based on the provided update function.
+   * @param {Function} updateFunction - The function that updates the search parameters.
+   */
+  const updateURLWithSearchParams = useCallback(
+    (updateFunction) => {
+      const searchParams = new URLSearchParams(history.location.search);
+      updateFunction(searchParams);
+      const url = decodeURIComponent(
+        history.location.pathname + '?' + searchParams.toString()
+      );
+      history.push(url);
+    },
+    [history]
+  );
+
   const toFlowchartPage = useCallback(() => {
-    const url = generatePath(routes.flowchart.main);
-    history.push(url);
-  }, [history]);
+    updateURLWithSearchParams(retainOtherQueryParams);
+  }, [updateURLWithSearchParams]);
 
   const toSelectedPipeline = useCallback(
     (pipelineValue) => {
-      // Get the value from param if it exists first
-      // before checking from localStorage
-      const activePipeline = pipelineValue
-        ? pipelineValue
-        : getCurrentActivePipeline();
-
-      const url = generatePath(routes.flowchart.selectedPipeline, {
-        pipelineId: activePipeline,
+      updateURLWithSearchParams((searchParams) => {
+        retainOtherQueryParams(searchParams);
+
+        // Get the value from param if it exists first
+        // before checking from localStorage
+        const activePipeline = pipelineValue
+          ? pipelineValue
+          : getCurrentActivePipeline();
+        searchParams.set(params.pipeline, activePipeline);
       });
-
-      history.push(url);
     },
-    [history]
+    [updateURLWithSearchParams]
   );
 
   const toSelectedNode = useCallback(
     (item) => {
-      const activePipeline = getCurrentActivePipeline();
-
-      const url = generatePath(routes.flowchart.selectedNode, {
-        pipelineId: activePipeline,
-        id: item.id,
+      updateURLWithSearchParams((searchParams) => {
+        const activePipeline = getCurrentActivePipeline();
+        searchParams.set(params.pipeline, activePipeline);
```

Comment:
what is the need to set active pipeline (params.pipeline) every time when the hook is called to set either selectedNode or focusedModularPipeline ? can't we just append/update the selectednode or focused mod pipeline to the existing searchParams (which should already have activePipeline), like we do in toSetQueryParam ?

Reply:
@ravi-kumar-pilla Good catch!
If you see existing behaviour on demo.kedro.org, we do not have `pipeline_id` params in url on initial load, it only added to url when you either click on focus button or on any node. So this was needed before stateful url changes. 
now as we already have `pipeline_id` params in url on initial load, so we don't need to set it along with either selectedNode or focusedModularPipeline ?